### DISCUSSION TITLE
How to get a row's source table name while querying its parent table?

### DISCUSSION THREAD

**Discussion by simonnelse** [sqlalchemy/sqlalchemy#8132] [Usage Questions] [answered]
To identify the source table of a particular row, We can use the [tableoid](https://www.postgresql.org/docs/current/ddl-system-columns.html) in PostgreSQL. A cast to [regclass](https://www.postgresql.org/docs/current/datatype-oid.html) retrieves the actual name of the table.

```sql
SELECT *, tableoid::regclass::text AS table_name
FROM   parent_tbl
WHERE  <some_condition>;
```

Is this achievable via SQLAlchemy? My requirement is to get the name of a row's source table when querying the parent it inherits from.

Thanks.

**Comment by simonnelse** [comment, marked as answer]
Was able to retrieve row's source table name like this -

```python
from sqlalchemy import column, func
from sqlalchemy.sql import select
from sqlalchemy.types import TEXT
from sqlalchemy.dialects.postgresql import REGCLASS

select(func.cast(func.cast(column('tableoid', is_literal=True), REGCLASS), TEXT))
```

**Reply by zzzeek** [reply]
that works, can you also cast as "tableoid::regclass" ?  we also do have an [OID](https://docs.sqlalchemy.org/en/14/dialects/postgresql.html?highlight=oid#sqlalchemy.dialects.postgresql.OID) datatype which might be compatible ?
